crates/cli-support/src/js/mod.rs

```diff
@@ -91,69 +60,48 @@ pub struct ExportedClass {
     wrap_needed: bool,
 }
 
-pub struct SubContext<'a, 'b: 'a> {
-    pub program: &'b decode::Program<'b>,
-    pub cx: &'a mut Context<'b>,
-    pub vendor_prefixes: HashMap<&'b str, Vec<&'b str>>,
-}
-
-pub enum ImportTarget {
-    Function(String),
-    Method(String),
-    Constructor(String),
-    StructuralMethod(String),
-    StructuralGetter(Option<String>, String),
-    StructuralSetter(Option<String>, String),
-    StructuralIndexingGetter(Option<String>),
-    StructuralIndexingSetter(Option<String>),
-    StructuralIndexingDeleter(Option<String>),
-}
-
-/// Return value of `determine_import` which is where we look at an imported
-/// function AST and figure out where it's actually being imported from
-/// (performing some validation checks and whatnot).
-enum Import<'a> {
-    /// An item is imported from the global scope. The `name` is what's imported
-    /// and the optional `field` is the field on that item we're importing.
-    Global {
-        name: &'a str,
-        field: Option<&'a str>,
-    },
-    /// Same as `Global`, except the `name` is imported via an ESM import from
-    /// the specified `module` path.
-    Module {
-        module: &'a str,
-        name: &'a str,
-        field: Option<&'a str>,
-    },
-    /// Same as `Module`, except we're importing from a local module defined in
-    /// a local JS snippet.
-    LocalModule {
-        module: &'a str,
-        name: &'a str,
-        field: Option<&'a str>,
-    },
-    /// Same as `Module`, except we're importing from an `inline_js` attribute
-    InlineJs {
-        unique_crate_identifier: &'a str,
-        snippet_idx_in_crate: usize,
-        name: &'a str,
-        field: Option<&'a str>,
-    },
-    /// A global import which may have a number of vendor prefixes associated
-    /// with it, like `webkitAudioPrefix`. The `name` is the name to test
-    /// whether it's prefixed.
-    VendorPrefixed {
-        name: &'a str,
-        prefixes: Vec<&'a str>,
-    },
-}
-
 const INITIAL_HEAP_VALUES: &[&str] = &["undefined", "null", "true", "false"];
 // Must be kept in sync with `src/lib.rs` of the `wasm-bindgen` crate
 const INITIAL_HEAP_OFFSET: usize = 32;
 
 impl<'a> Context<'a> {
+    pub fn new(module: &'a mut Module, config: &'a Bindgen) -> Result<Context<'a>, Error> {
+        // Find the single memory, if there is one, and for ease of use in our
+        // binding generation just inject one if there's not one already (and
+        // we'll clean it up later if we end up not using it).
+        let mut memories = module.memories.iter().map(|m| m.id());
+        let memory = memories.next();
+        if memories.next().is_some() {
+            bail!("multiple memories currently not supported");
+        }
+        drop(memories);
```

Comment:
I think it's because `memories` is an `impl Iterator` which means rustc pessimistically assumes it has a dtor which should extend the lifetime. That's at least my best rationalization for the compile failure if this line is removed :)